src/erlfmt.erl

```diff
@@ -64,18 +69,35 @@ init(State) ->
     rebar3_fmt_prv:init(State).
 
 %% API entry point
--spec format_file(file:name_all(), out()) -> {ok, [error_info()]} | {error, error_info()}.
-format_file(FileName, Out) ->
+-spec format_file(file:name_all(), config()) -> {ok, [error_info()]} | {error, error_info()}.
+format_file(FileName, {Pragma, Out}) ->
     try
         {ok, Forms, Warnings} = file_read_forms(FileName),
-        [$\n | Formatted] = format_forms(Forms),
-        verify_forms(FileName, Forms, Formatted),
-        write_forms(FileName, Formatted, Out),
-        {ok, Warnings}
+        ShouldFormat = (Pragma == ignore) or contains_pragma_forms(Forms),
```

Comment:
This is good progress, but it still parses the whole file just to check the first couple lines. Since in `file_read_forms` we special-case the first form already (by calling `io_scan` before entering the main loop), I was thinking we could check this in there and skip parsing the rest of the file if we don't detect the pragma. I think it would be reasonable to return `ignore` from this function if that happens.

Reply:
As an addition, please avoid using `and` and `or` in favour of `andalso` and `orelse`. The short forms are eager, so `or` evaluates the right side even if the left side is false.

Reply:
Oh `orelse ` that is very interesting.

Ah yes, `file_read_forms`, great idea.

Reply:
These are both addressed in newest commit, I hope.